src/GafferOSL/OSLShader.cpp

```diff
@@ -339,19 +339,25 @@ Plug *loadNumericArrayParameter( const OSLQuery::Parameter *parameter, const Int
 {
 	typedef typename PlugType::ValueType DataType;
 	typedef typename DataType::ValueType ValueType;
+	typedef typename ValueType::value_type ElementType;
 
-	ValueType defaultValue;
+	typename DataType::Ptr defaultValueData = new DataType();
 	if( parameter->idefault.size() )
 	{
-		defaultValue.resize( parameter->idefault.size() );
-		std::copy( parameter->idefault.begin(), parameter->idefault.end(), defaultValue.begin() );
+		defaultValueData->writable().resize( parameter->idefault.size() );
+		for( size_t i = 0; i < parameter->idefault.size(); i++ )
+		{
+			defaultValueData->writable()[i] = ElementType( parameter->idefault[i] );
```

Comment:
I know this isn't the most performance critical part of the code, but calling `writable()` does have overhead, and I think it's worth not being in the habit of calling it inside loops like this. Could you switch to this style throughout?

```
typename DataType::Ptr defaultValueData = new DataType();
ValueType &defaultValue = defaultValueData->writable();
// Do work on defaultValue here
```